src/main/java/it/systems/paymentreviser/entity/Case.java

```diff
@@ -0,0 +1,51 @@
+package it.systems.paymentreviser.entity;
+
+import it.systems.paymentreviser.dto.PaymentDTO;
+import it.systems.paymentreviser.enums.ResolutionStatus;
+import it.systems.paymentreviser.enums.ValidCurrency;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+public abstract class Case {
+	
+	protected Integer id;
+	protected ResolutionStatus resolution;
+	protected final Payment payment;
+
+	protected Case(PaymentDTO paymentDTO) {
```

Comment:
By accepting a DTO here, the Case gets coupled to it, and through it to the controller indirectly, because the DTO belongs to the controller (a different controller would have different DTOs). Therefore a two-way dependency is introduced: from case to DTO/controller and from DTO/controller to case.

A better way would be for this constructor to take `Payment`, which would be constructed in controller.